docs/architecture/adr-104-out-of-band-state-sync.md

```diff
@@ -124,58 +138,173 @@ in order to adopt this solution.
+or trigger snapshot creation using `export`. 
+
+The client side, restores the application state from a local snapshot that was previously exported, using
+`restore`. Upon successful completion of the previous sequence of commands, the state of CometBFT is bootstrapped
+using `bootstrap-state` and CometBFT can be launched. 
+
+
+There are three prerequisites for this solution to work when a node is syncing:
+1. The application has access to the snapshot database
+2. CometBFTs state and block stores are empty or reset
+3. CometBFT is not running while the node is state syncing 
+
+The server side of state sync (snapshot generation and dumping), can be performed while the node is running.
+The application has to be careful not to interfere with normal node operations, and to use a snapshot store
+and dumping mechanism that will mitigate the risk of requesting snapshots while they are being dumped to an archive format. 
+
+In order to be able to dump or export the snapshots, the application must have access to the snapshot store. 
+
+We describe the main interface expected from the snapshot database and used by the above mentioned CLI commands
+for snapshot manipulation. 
+The interface was derived from the SDK's implementation of local state sync.  
+
+```golang
+
+// Delete a snapshot for a certain height
+func (s *Store) Delete(height uint64, format uint32) error
+
+// Retrieves a snapshot for a certain height
+func (s *Store) Get(height uint64, format uint32) (*types.Snapshot, error) 
+
+// List recent snapshots, in reverse order (newest first)
+func (s *Store) List() ([]*types.Snapshot, error)
+
+// Loads a snapshot (both metadata and binary chunks). The chunks must be consumed and closed.
+// Returns nil if the snapshot does not exist.
+func (s *Store) Load(height uint64, format uint32) (*types.Snapshot, <-chan io.ReadCloser, error)
+
+// LoadChunk loads a chunk from disk, or returns nil if it does not exist. The caller must call
+// Close() on it when done.
+func (s *Store) LoadChunk(height uint64, format, chunk uint32) (io.ReadCloser, error)
+
+// Save saves a snapshot to disk, returning it.
+func (s *Store) Save(height uint64, format uint32, chunks <-chan io.ReadCloser) (*types.Snapshot, error) 
+
+// PathChunk generates a snapshot chunk path.
+func (s *Store) PathChunk(height uint64, format, chunk uint32) string
+
+```
+
+In order to dump a snapshot, an application needs to retrieve all the chunks stored at a certain path. 
+
+**CometBFT state bootstrap**
+
+In addition to managing snapshots, it is neccessary to bootstrap (setup) the state and block store of CometBFT
+in order for CometBFT to continue with block sync and consensus after state sync. 
+At the moment of writing this ADR, there is no command line in CometBFT that supports this.
+Thus application developers have to, within their bootstraping command:
+
+- Create a state and block store
+- Launch a light client to verify the snapshot headers
```

Comment:
Not the headers. The light client is launched to learn the block header for the height of the snapshot.